extensions/reactive-mysql-client/runtime/src/main/java/io/quarkus/reactive/mysql/client/runtime/MySQLPoolProducer.java

```diff
@@ -4,36 +4,90 @@
 import javax.enterprise.inject.Produces;
 import javax.inject.Singleton;
 
+import org.jboss.logging.Logger;
+
 import io.vertx.mysqlclient.MySQLPool;
 
 @ApplicationScoped
 public class MySQLPoolProducer {
 
+    private static final Logger LOGGER = Logger.getLogger(MySQLPoolProducer.class);
+
     private volatile MySQLPool mysqlPool;
-    private volatile io.vertx.axle.mysqlclient.MySQLPool axleMySQLPool;
-    private volatile io.vertx.reactivex.mysqlclient.MySQLPool rxMySQLPool;
+    private io.vertx.mutiny.mysqlclient.MySQLPool mutinyMySQLPool;
```

Comment:
How about using `AtomicReference` to remove synchronization without losing the checks in the producer methods?

Reply:
`AtomicReference.compareAndSet(null, instance)` requires the instance to be already created - so it's not lazy. Using `updateAndGet` could work but is rather convoluted (you would need to move the `if` check in the binary operator).

Reply:
Ok, let's revisit later if that proves to be a problem